docs/nuget-sdk.md

```diff
@@ -0,0 +1,85 @@
+# NuGet SDK
+
+NuGet is primarily a tool, but we also publish packages that others can reference in their own projects. All of these packages are published to NuGet.org with the [`NuGet.` prefix](https://www.nuget.org/packages?q=nuget.*). Documentation on the NuGet SDK is on our docs site [https://docs.microsoft.com/nuget/reference/nuget-client-sdk](https://docs.microsoft.com/nuget/reference/nuget-client-sdk)
+
+The projects that are part of the SDK are largely under `src\NuGet.Core`, but also include `NuGet.VisualStudio` and `NuGet.VisualStudio.Contracts` under `src\NuGet.Clients`, which define the interfaces for our Visual Studio extensibility APIs.
+
+## API compatibility policy
+
+NuGet is primarily a tool, and as such our efforts are focused on: Visual Studio integration, `dotnet` cli integration, `msbuild` integration, and `nuget.exe`.
+
+When we need to break APIs part of the SDK to meet our tooling needs, we will do so. However, we understand this negatively impacts customers using the SDK, so we will try to avoid doing so by trying to be attentive in code reviews, and using tooling such as the [PublicApiAnalyzers](https://github.com/dotnet/roslyn-analyzers/tree/master/src/PublicApiAnalyzers).
```

Comment:
Why not marking APIs as obsolete first and then eliminating later?

Reply:
Ideally we should, but if you mark something as obsolete while also breaking the API, it's no better than just deleting the API since it will still fail at runtime. In fact, failing to compile gives customers using the SDK faster feedback that it doesn't work.

If you're reviewing a PR where there are changes to `PublicAPI.Shipped.txt`, you should ask the PR author exactly that question.